code/datums/station_traits/job_traits.dm

```diff
@@ -7,13 +11,22 @@
 	trait_flags = STATION_TRAIT_ABSTRACT
 	/// What tooltip to show on the button
 	var/button_desc = "Sign up to gain some kind of unusual job, not available in most rounds."
+	/// Can this job roll antag?
+	var/can_roll_antag = CAN_ROLL_ALWAYS
+	/// How many positions to spawn?
+	var/position_amount = 1
 	/// Type of job to enable
-	var/job_to_add = /datum/job/clown
+	var/datum/job/job_to_add = /datum/job/clown
 	/// Who signed up to this in the lobby
 	var/list/lobby_candidates
 
 /datum/station_trait/job/New()
 	. = ..()
+	switch(can_roll_antag)
+		if(CAN_ROLL_PROTECTED)
+			SSstation.antag_protected_roles += job_to_add::title
+		if(CAN_ROLL_NEVER)
+			SSstation.antag_restricted_roles += job_to_add::title
```

Comment:
we're sure this works? i thought `::` was always calculated at compile time. this feels like it's going to give you the initial of `/datum/job::title`, not of whatever the job is. Double check this.

Reply:
![Uploading image.png…]()

Reply:
its essentially just a wrapper for `initial(job_to_add.title)`

Reply:
we need to decide which one of these we want